src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -613,6 +613,121 @@ A list of all generated pages. It is an object with one property.
 
 - `pathname` - the finalized path of the page.
 
+### Defining custom hooks
+
+As an integration author, you can define custom hooks that your integration will call. This allows for runtime interactions between your integration and other integrations. Other integrations can provide implementations for your custom hooks and your integration can call those at the appropriate moment in your integration's lifecycle.
+
+#### Defining hook types
+
+Provide the type for your custom hook by augmenting Astro’s `IntegrationHooks` interface within your integration:
+
+```ts title="types.ts"
+// on an implementation file
+declare global {
+  namespace Astro {
+    interface IntegrationHooks {
+      'myLib:eventHappened': (your: string, parameters: number) => Promise<void>;
+    }
+  }
+}
+```
+
+#### Calling custom hooks
+
+Installed integrations can be accessed from the `astro:config:setup` and `astro:config:done` hooks. Loop over these to call your custom hook on each if available:
+
+```ts {7-9}
+// your-integration/index.ts
+export default (): AstroIntegration => {
+  return {
+    name: 'your-integration',
+    hooks: {
+      'astro:config:setup': async ({ config }) => {
+        for (const integration of config.integrations) {
+          await integration.hooks['myLib:eventHappened'].?('your values', 123);
+        }
+      },
+    }
+  }
+}
+```
+
+To call custom hooks outside of `astro:config:setup` and `astro:config:done`, store the integrations array and use it later. `astro:config:done` should be preferred to work with integrations that get added by other integrations during `astro:config:setup`.
```

Comment:
This actually feels like it contradicts the previous part, which honestly made it sound like your hook happens "tied to" or "at" setup or done (only).

Also, I think instead of "outside of" setup/done, maybe instead referring to "another hook", because from the example it looks like you are using this by tapping in to a different hook.

So, I'm left wondering why the section below told me things could be accessed by two hooks (which I will take as an exhaustive list), then this part shows me using a different Astro hook.

Reply:
Using the same example as the other comment:

```ts
defineConfig({
  integrations: [
    yourIntegration(),
    mdx(),
  ],
});
```

If `mdx` calls `updateConfig` during `astro:config:setup` to add a `tsx` integration, that new integration will only be visible to `yourIntegration` on `astro:config:done`. So, if your custom hook happens during your own `astro:config:setup` there is nothing to do, it happens before the other integration is even present to react to it. But if that is not the case, it is better to use `astro:config:done` so all integrations can react

Reply:
Right, this is more a note of the actual text wording here! The paragraph before sounded like "this is how you do it" and this sentence sounded like, "To do it another way" which felt a little jarring.

So this is all great stuff, I just am at the same time doing any combination of confirming some things that I honestly don't have experience with, and also pointing out where generic text/language use feels a little off or out of flow. This note was more about massaging the text a bit more to remove that feeling, and the extra context helps!